luarules/gadgets/unit_minesweeper_detection.lua

```diff
@@ -5,201 +5,119 @@ function gadget:GetInfo()
+			revealedMines[currentTeam][mineToReveal] = true
+			--show in vision and radar, and prevent engine from immediately resseting that state
+			Spring.Echo("revleaed", mineToReveal, "to", currentTeam)
+			Spring.SetUnitLosState(mineToReveal, teamIDs[currentTeam], 3)
+			Spring.SetUnitLosMask(mineToReveal, teamIDs[currentTeam], 3)
+		end
+	end
+end
 
-		
-
-			for team in pairs(teamIDs) do
-
-				--Spring.Echo('hornet minesToReveal (team)' , team)
-				--Spring.Debug.TableEcho(minesToReveal)
-				for mineToReveal in next, (minesToReveal[team] or {}) do
-					--Spring.Echo('hornet poi2 minesToReveal loop starting, team: ' , team)
-					--Spring.Echo('mineToReveal loop,', mineToReveal)
-					if revealedMines[team][mineToReveal] == nil then
-						--show 
-						--Spring.Echo('revealedMines[team][mineToReveal] = mineToReveal', team, mineToReveal)
-						revealedMines[team][mineToReveal] = mineToReveal
-						Spring.SetUnitLosState(mineToReveal, team, 3) --show in vision and radar
-						Spring.SetUnitLosMask(mineToReveal, team, 3) -- prevent engine from immediately resetting that state
-
-					end
-				end
-					
-
-				--Spring.Echo('hornet revealedMines (team)' , team)
-				--Spring.Debug.TableEcho(revealedMines)
-				if next(revealedMines[team]) ~= nil then
-
-					--Spring.Echo('hornet minesToReveal (team)' , team)
-					--Spring.Debug.TableEcho(minesToReveal)
-
-					--if any mines -are- uncloaked that no longer should be, recloak
-					--if table.count(minesToReveal[team]) > 0 then
-					if next(minesToReveal[team]) ~= nil then
-
-						for revealedMine in pairs(revealedMines[team]) do
-							if minesToReveal[team][revealedMine] == nil then
-								--reset this team mask
-								Spring.SetUnitLosMask(revealedMine, team, 0)
-								revealedMines[team][revealedMine] = nil
-							end
-						end
-					else
-						--if empty and previously decloaked not empty, reset all mines
-						for revealedMine in pairs(revealedMines[team]) do
-							Spring.SetUnitLosMask(revealedMine, team, 0)
-							revealedMines[team][revealedMine] = nil
-						end
-					end
-
-				end
+local function processRevealedMines(teamID)
+	for revealedMine, _ in pairs(revealedMines[teamID]) do
+		Spring.SetUnitLosMask(revealedMine, teamID, 0)
+		revealedMines[teamID][revealedMine] = nil
+	end
+end
 
+-- if we have more teams than frames per second we use a slightly slower method
+local GAMESPEED = Game.gameSpeed
+if #teamIDs + 1 >= GAMESPEED then
+	function gadget:GameFrame(f)
+		local loopedFrame = f % GAMESPEED
+		for teamID in pairs(teamIDs) do
+			if teamID % GAMESPEED == loopedFrame then
+				processRevealedMines(teamID)
+				processesTeamsSweepers(teamID)
 			end
-
-		end--end frame check
-
+		end
 	end
-
-end
+else
+	function gadget:GameFrame(f)
+		local currentTeam = f%GAMESPEED
+		if currentTeam > #teamIDs then
+			return
+		end
+
+		processRevealedMines(currentTeam)
+		processesTeamsSweepers(currentTeam)
+	end
```

Comment:
```suggestion
function gadget:GameFrame(n)
	for teamId in pairs(teamIds) do
		if teamId % processingInterval == 0 then
			processRevealedMines(currentTeam)
			processesTeamsSweepers(currentTeam)
		end
	end
end
```
Why complicate it? This evenly distributes processing all teams over the interval, no need to check how many teams there are.

(Also remove the check and other gameFrame callin, Github doesn't allow code suggestions across deleted lines.)